csrc/host_ir/lower_to_llvm.cpp

```diff
@@ -0,0 +1,912 @@
+            output_outer_val->toString() + "_split_extent"
+          );
+        }
+        else{
+          if(split_expr->factor()->isConstInt()){
+            val2stride_map[graph.toGroup(output_outer_val)].llvm_extent = builder.getInt64(split_factor);
+          }
+          else{
+            if(val2stride_map.find(graph.toGroup(split_expr->factor())) != val2stride_map.end()){
+              val2stride_map[graph.toGroup(output_outer_val)].llvm_extent = val2stride_map[graph.toGroup(split_expr->factor())].llvm_extent;
+            }
+            else{
+              NVF_ERROR(false, "LLVM Lowering Error: Outer split factor is not a constant and not found in val2stride_map");
+              return;
+            }
+          }
+          if(val2stride_map[graph.toGroup(input_val)].llvm_extent == nullptr || val2stride_map[graph.toGroup(output_inner_val)].llvm_extent == nullptr || val2stride_map[graph.toGroup(output_outer_val)].llvm_extent != nullptr){
+            return;
+          }
+          val2stride_map[graph.toGroup(output_inner_val)].llvm_extent = builder.CreateUDiv(
+            val2stride_map[graph.toGroup(input_val)].llvm_extent,
+            val2stride_map[graph.toGroup(output_outer_val)].llvm_extent,
+            output_inner_val->toString() + "_split_extent"
+          );
+        }
+
+    } else { // Fallback for other ops (e.g., simple unary pass-through)
+        NVF_ERROR(false, "LLVM Lowering Error: Unhandled op_type '" + op_type + "' for Val " + current_val_to_process->toString());
+    }
+}
+
+/*
+
+Generate infer stride module
+
+*/
+llvm::orc::ThreadSafeModule generate_infer_stride_module(std::vector<IterDomain*>& logical_domain, std::vector<IterDomain*>& allocation_domain, Fusion& fusion, std::string name) {
+  auto Context = std::make_unique<llvm::LLVMContext>();
+  auto* ctx = Context.get();
+  auto Module = std::make_unique<llvm::Module>(name, *ctx);
+  llvm::IRBuilder<> builder(*ctx);
+  auto* int64Ty = llvm::Type::getInt64Ty(*ctx);
+  auto* ptrTy = llvm::PointerType::getUnqual(int64Ty);
+
+  auto* funcTy = llvm::FunctionType::get(llvm::Type::getVoidTy(*ctx), {ptrTy, int64Ty, ptrTy, int64Ty, ptrTy, int64Ty}, false);
+  auto* func = llvm::Function::Create(funcTy, llvm::Function::ExternalLinkage, "infer_stride", Module.get());
+  auto* entry = llvm::BasicBlock::Create(*ctx, "entry", func);
+  builder.SetInsertPoint(entry);
+
+  std::vector<Val*> input_vals = domain2vals(logical_domain);
+  std::vector<Val*> output_vals = domain2vals(allocation_domain);
+  auto arg_it = func->arg_begin();
+  llvm::Value* input_shape_buffer_ptr = &*arg_it;
+  llvm::Value* output_stride_buffer_ptr = &*arg_it+2;
+  llvm::Value* output_shape_buffer_ptr = &*arg_it+4;
+  std::unordered_map<ValGroup, StrideInfo> val2stride;
+  std::unordered_map<int, Val*> boundary_vals;
+  for(size_t i = 0; i < input_vals.size(); i++){
+    boundary_vals[i] = input_vals[i];
+  }
+
+  IdModel id_model(&fusion);
```

Comment:
I suspect IdModel buys you little. For top-level expressions (e.g. allocation), we mostly care about extents, which are mapped across ops by construction, not indices. I also doubt we exercised IdModel at all on non-SSA IR (e.g. host IR and kernel IR).